ui/imports/shared/views/AmountToSend.qml

```diff
@@ -17,7 +17,7 @@ ColumnLayout {
     property var selectedAsset
     property bool isBridgeTx: false
     property bool interactive: false
-    property var maxFiatBalance
```

Comment:
I'd rather pass the object instead of the plain double. If we ever need the string representation of `maxFiatBalance` inside this component you'll need the extra info.
It also looks to me that the code before and after the changes should be functionally equivalent. Was there some spot where the object was passed and the float value was needed that I'm missing?

Reply:
Hey so what was happening we had a `var` type which pointed to `CurrencyBalance` type. and when model was reset it pointed to null and caused the issue we saw.

so now with what I have done the values are getting update correctly as I dont have anything that points to CurrencyBalance object. however I do feel this is a workaround and would be sorted if we used C++ instead. Cause this behavior is not expected.

Reply:
Oh I see... Now that I look better, it seems we make the check for null everywhere except for line 95 of `AmountToSend.qml`. Perhaps that's the only needed change?

Reply:
nope the issue was the evaluation of the maxFiatBalance value which was displayed on top as "balance exceeded"

That was not getting recalculated